java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/PostgresValidator.java

```diff
@@ -652,55 +666,144 @@ public void close() throws Exception {
+                    return false;
+                }
             case "smallint":
-                return Data.DataType.TypeName.INT16_VALUE <= val
-                        && val <= Data.DataType.TypeName.INT64_VALUE;
+            case "smallserial":
+                // SMALLINT, SMALLSERIAL -> SMALLINT
+                return val == Data.DataType.TypeName.INT16_VALUE;
             case "integer":
-                return Data.DataType.TypeName.INT32_VALUE <= val
-                        && val <= Data.DataType.TypeName.INT64_VALUE;
+            case "serial":
+                // INTEGER, SERIAL -> INTEGER
+                return val == Data.DataType.TypeName.INT32_VALUE;
             case "bigint":
+            case "bigserial":
+            case "oid":
+                // BIGINT, BIGSERIAL, OID -> BIGINT
                 return val == Data.DataType.TypeName.INT64_VALUE;
-            case "float":
             case "real":
+                // REAL -> REAL
                 return val == Data.DataType.TypeName.FLOAT_VALUE;
-            case "boolean":
-                return val == Data.DataType.TypeName.BOOLEAN_VALUE;
             case "double":
             case "double precision":
+                // DOUBLE PRECISION -> DOUBLE PRECISION
                 return val == Data.DataType.TypeName.DOUBLE_VALUE;
-            case "decimal":
-            case "numeric":
-                return val == Data.DataType.TypeName.DECIMAL_VALUE
-                        // We allow user to map numeric into rw_int256 or varchar to avoid precision
-                        // loss in the conversion from pg-numeric to rw-numeric
-                        || val == Data.DataType.TypeName.INT256_VALUE
-                        || val == Data.DataType.TypeName.VARCHAR_VALUE;
+            case "char":
+            case "character":
             case "varchar":
             case "character varying":
+            case "xml":
             case "uuid":
+            case "text":
+            case "inet":
+            case "cidr":
+            case "macaddr":
+            case "macaddr8":
+            case "int4range":
+            case "int8range":
+            case "numrange":
+            case "tsrange":
+            case "tstzrange":
+            case "daterange":
             case "enum":
+                // CHARACTER, CHARACTER VARYING, VARCHAR, CHAR, XML, UUID, CITEXT, INET, CIDR,
+                // MACADDR, MACADDR8, INT4RANGE, INT8RANGE, NUMRANGE, TSRANGE, TSTZRANGE, DATERANGE,
+                // ENUM -> CHARACTER VARYING
                 return val == Data.DataType.TypeName.VARCHAR_VALUE;
+            case "timestamptz":
+            case "timestamp with time zone":
+                // TIMESTAMPTZ, TIMESTAMP WITH TIME ZONE -> TIMESTAMP WITH TIME ZONE
+                return val == Data.DataType.TypeName.TIMESTAMPTZ_VALUE;
+            case "timetz":
+            case "time with time zone":
+            case "time without time zone":
```

Comment:
Why is `time without time zone` here rather than together with its shorter form `time` below?

To be honest, if this `isDataTypeCompatible` is only used to process data from `data_type` column of `information_schema.columns`, it will NEVER see `varchar` and `character varying` at the same time - only a single form would be used by PostgreSQL. The current list is neither precise nor comprehensive.

Reply:
Indeed, thanks for pointing out.